include/deal.II/grid/manifold_lib.h

```diff
@@ -582,10 +602,36 @@ class FunctionManifold : public ChartManifold<dim, spacedim, chartdim>
    * pointers.
    */
   const bool owns_pointers;
+
+  /**
+   * The expresssion used to construct the push_forward function.
+   */
+  const std::string push_forward_expression;
```

Comment:
I guess you have to add these members because you can't clone the `ParsedFunction` object that's behind the `push_forward_function`/`pull_back_function`? Could you avoid this if you simply made them `std::shared_ptr` objects? Then every clone of this mapping would simply be using the same function object.

Reply:
We have a constructor that takes references to `Function` objects. In this case, the internal pointers point to areas of memory that we do not own, and we can't just pass around shared_ptr. This would lead to double free problems if the first type of constructor is used.

Reply:
Couldn't we somehow do as follows:
* If `owns_pointer==false`, then we simply copy the pointer to the cloned object and set its `owns_pointer` variable also to `false`.
* If `owns_pointer==true`, then we know that the object in question is in fact a `FunctionParser` object. If we had a way to either clone that, or to use a `std::shared_ptr` to it that we could share between the current `FunctionManifold` and the cloned object, then we wouldn't have to keep all of those additional variables around.

Reply:
It is possible, but I have limited amount of time, and I'm trying to fix the tests of the other PR. I think this is a minor issue. If you really want this to be addressed before 9.0, open an issue and I'll get back to this later on.

Reply:
I do not think this is major as it is in the private section of the class, so it can wait in my opinion.

Reply:
I agree.

@luca-heltai : (i) is this tested anywhere? (ii) I'm traveling and I don't want to forget it. If it's getting merged, would you mind creating the reminder github issue?

Reply:
See #6214